typescript/lib/config.ts

```diff
@@ -822,5 +823,29 @@ export class AIConfigRuntime implements AIConfig {
     return this.metadata.models?.[modelName];
   }
 
+  /**
+   * Generates a ModelMetadata object based on the provided inference settings and model ID.
+   *
+   * @param inferenceSettings - The inference settings to be used for the model.
+   * @param modelId - The unique identifier for the model.
+   * @returns A ModelMetadata object that includes the model's name and optional settings.
+   */
+  public generateModelMetadata(
```

Comment:
Maybe `mergeModelMetadata`?

Reply:
While this method is merging settings, its primary purpose is to generate the modelMetadata object. The act of merging is a part of the process but I think it doesn't define the overall purpose of the function. Could we maybe do something like `getOverridesAndCreateModelMetadata`?

Updated docstring to reflect merging & override extraction:
`Generates a ModelMetadata object from the inference settings by extracting the settings that override the global settings.`

Reply:
My issue is that even as a creator of the library I don't understand what this function does, so it won't be clear to any user of the library either. Maybe just `getModelMetadata` then.

Reply:
Updated to `getModelMetadata`. Can comeback to this if we think of something better.